pkg/istio-agent/xds-proxy.go

```diff
@@ -42,14 +43,27 @@ import (
 	"istio.io/istio/pilot/pkg/dns"
 	nds "istio.io/istio/pilot/pkg/proto"
 	v3 "istio.io/istio/pilot/pkg/xds/v3"
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/pkg/filewatcher"
 	"istio.io/pkg/log"
 )
 
+var (
+	newFileWatcher = filewatcher.NewWatcher
+)
+
 const (
 	defaultClientMaxReceiveMessageSize = math.MaxInt32
-	defaultInitialConnWindowSize       = 1024 * 1024     // default gRPC InitialWindowSize
-	defaultInitialWindowSize           = 1024 * 1024     // default gRPC ConnWindowSize
-	sendTimeout                        = 5 * time.Second // default upstream send timeout.
+	defaultInitialConnWindowSize       = 1024 * 1024            // default gRPC InitialWindowSize
+	defaultInitialWindowSize           = 1024 * 1024            // default gRPC ConnWindowSize
+	sendTimeout                        = 5 * time.Second        // default upstream send timeout.
+	watchDebounceDelay                 = 100 * time.Millisecond // file watcher event debounce delay.
+)
+
+const (
+	MetadataClientCertKey   = "ISTIO_META_TLS_CLIENT_KEY"
```

Comment:
are these used *only* for XDS? Or for CSR as well?

Reply:
It is not used for CSR - BTW, can you point me to that code - I am not aware of that code

Reply:
https://github.com/istio/istio/blob/2c81d37a8627ed2b880ee8dd961f7982b94ab1d2/security/pkg/nodeagent/caclient/providers/citadel/client.go#L137